check_pgactivity

```diff
@@ -5600,6 +5604,167 @@ sub check_oldest_idlexact {
 }
 
 
+=item B<oldest_xmin> (8.4+)
+
+Check the xmin I<horizon> per distinct sources of xmin retention.
+
+Perfdata contains the oldest xmin and maximum age for the following source of
+xmin retention: C<query> (a currently executing query), C<active_xact> (an
+opened transaction currently executin a query), C<idle_xact> (an opened
+transaction being idle), C<2pc> (a pending prepared transaction), C<repslot> (a
+recpliation slot) and C<walwender> (a WAL sender replication process).  If a
+source doesn't retain any transaction, NaN is returned.
+For versions prior to 9.4, only C<2pc> source of xmin retention is available,
+so other sources won't appear in the perfdata.
+
+Critical and Warning thresholds only accept a raw number of transaction.
+
+Required privileges: an unprivileged role checks only its own queries;
+a pg_read_all_stats (10+) or superuser (<10) role is required to check
+pg_stat_replication.  2PC, pg_stat_activity, and replication slots don't
+require special privileges.
+
+=cut
+
+sub check_oldest_xmin {
+    my @rs;
+    my @perfdata;
+    my @msg;
+    my @msg_crit;
+    my @msg_warn;
+    my @hosts;
+    my $c_limit;
+    my $w_limit;
+    my %args        = %{ $_[0] };
+    my $me          = 'POSTGRES_OLDEST_XMIN';
+    my %queries     = (
+        # 8.4 is the first supported version as we rely on window functions to
+        # get the oldest xmin.  Only 2PC has transaction information available
```

Comment:
Probably yes, although I'm not sure it's worth the trouble.

Reply:
It will be the same query for all version down to 8.1:

~~~
postgres@postgres=# SELECT version();
 PostgreSQL 8.1.23 on x86_64-unknown-linux-gnu, compiled by GCC gcc (Debian 8.3.0-6) 8.3.0

postgres@postgres=# \d pg_prepared_xacts 
 transaction | xid                      | 
 gid         | text                     | 
 prepared    | timestamp with time zone | 
 owner       | name                     | 
 database    | name                     | 
~~~

Reply:
well, ok, I give up on that as well.